ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -394,10 +391,242 @@ func TestVirtual_CallMethod_On_DeactivatedDirtyState(t *testing.T) {
+					CallType:            payload.CTMethod,
+					CallFlags:           payload.BuildCallFlags(deactivateIsolation.Interference, deactivateIsolation.State),
+					Caller:              server.GlobalCaller(),
+					Callee:              objectRef,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Deactivate",
+					CallOutgoing:        outgoing,
+				}
+				server.SendPayload(ctx, &pl)
+				commonTestUtils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
+				server.IncrementPulseAndWaitIdle(ctx)
+				synchronizeExecution.WakeUp()
+				commonTestUtils.WaitSignalsTimed(t, 10*time.Second, oneExecutionEnded)
+				commonTestUtils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+			}
+			require.Equal(t, 1, typedChecker.VStateReport.Count())
+			require.Equal(t, 1, typedChecker.VStateRequest.Count())
+			require.Equal(t, 1, typedChecker.VDelegatedRequestFinished.Count())
+			require.Equal(t, 1, typedChecker.VCallResult.Count())
+			typedChecker.MinimockFinish()
+		}
+
+		{
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+				return false
+			})
+			typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+				require.Equal(t, objectRef, report.Object)
+				require.Equal(t, payload.Inactive, report.Status)
+				require.True(t, report.DelegationSpec.IsZero())
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Nil(t, report.ProvidedContent)
+				return false
+			})
+			{ // setup runner mock for check call
+				runnerMock.AddExecutionClassify(checkMethodName, testCase.isolation, nil)
+				requestResult := requestresult.New([]byte(checkMethodName+" result"), objectRef)
+
+				if testCase.isolation.State == contract.CallValidated {
+					runnerMock.AddExecutionMock(checkMethodName).AddStart(
+						func(ctx execution.Context) {
+							require.Equal(t, objectRef, ctx.Request.Callee)
+							require.Equal(t, []byte(origDirtyMem), ctx.ObjectDescriptor.Memory())
+						},
+						&execution.Update{
+							Type:   execution.Done,
+							Result: requestResult,
+						},
+					)
+				}
+			}
+
+			{
+				pl := payload.VCallRequest{
+					CallType:            payload.CTMethod,
+					CallFlags:           payload.BuildCallFlags(testCase.isolation.Interference, testCase.isolation.State),
+					Caller:              server.GlobalCaller(),
+					Callee:              objectRef,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Check",
+					CallOutgoing:        server.BuildRandomOutgoingWithPulse(),
+				}
+				server.SendPayload(ctx, &pl)
+			}
+
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+
+			// increment pulse twice for stop SMStateFinalizer
+			server.IncrementPulseAndWaitIdle(ctx)
```

Comment:
Why we can't wait for SMStateFinalizer using WaitSignalsTimed ?

Reply:
We are already do this:
```
stateReportSend := server.Journal.WaitStopOf(&finalizedstate.SMStateFinalizer{}, 1)
...
// increment pulse twice for stop SMStateFinalizer
server.IncrementPulseAndWaitIdle(ctx)
server.IncrementPulseAndWaitIdle(ctx)

commonTestUtils.WaitSignalsTimed(t, 10*time.Second, stateReportSend)
```
But we need to send pulse twice because first one used for send report and create SMStateFinalizer and next one used for stop SMStateFinalizer.